exec/java-exec/src/main/java/org/apache/drill/exec/store/ischema/InfoSchemaGroupScan.java

```diff
@@ -111,12 +114,8 @@ public GroupScan clone(List<SchemaPath> columns) {
     return new InfoSchemaGroupScan(this);
   }
 
-  public void setFilterPushedDown(boolean status) {
-    this.isFilterPushedDown = status;
-  }
-
   @JsonIgnore
   public boolean isFilterPushedDown() {
-    return isFilterPushedDown;
+    return filter != null;
```

Comment:
This is tricky. This flag is probably acting more like "has filter push-down analysis been done": it prevents Calcite from pushing filters down multiple times (which won't work if the group scan removes the original `WHERE` clause conditions); and it prevents Calcite from needlessly repeating the analysis multiple times when there are no filters.

The proposed change means we will do that no-filter analysis repeatedly.

(As it turns out, I made this same mistake myself in a recent project.)

Reply:
I have removed this flag since instead of using additional boolean variable (which by the way was not serialized), it can be replaced with check if filter is set or not. Filter is set only if it was pushed down, so thus checking that filter was set `will prevent Calcite from pushing filters down multiple times`.

https://github.com/apache/drill/blob/master/exec/java-exec/src/main/java/org/apache/drill/exec/store/ischema/InfoSchemaPushFilterIntoRecordGenerator.java#L94-L99

Reply:
Sounds good.